tests/test_updown.py

```diff
@@ -4,144 +4,243 @@
+                ctadata.fetch_and_save_file(
+                    entry['href'], chunk_size=1024*1024*10)
+                assert 'in 4 chunks' in caplog.text
+                assert 'in 9 chunks' not in caplog.text
+                ctadata.fetch_and_save_file(
+                    entry['href'], chunk_size=1024*1024*5)
+                assert 'in 9 chunks' in caplog.text
+                break
+
+
+@pytest.mark.timeout(30)
+def test_apiclient_upload_single_file(start_service, caplog):
+    with webdav_server():
+        import ctadata
+        ctadata.APIClient.downloadservice = start_service['url']
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            subprocess.check_call([
+                "dd", "if=/dev/random", f"of={tmpdir}/local-file-example",
+                "bs=1M", "count=100"
+            ])
 
-def test_fetch(client: Any):
-    r = client.get(url_for('fetch', path="md5sum-lst.txt"))
-    assert r.status_code == 200
-    print(r.json)
+            r = ctadata.upload_file(f'{tmpdir}/local-file-example',
+                                    'example-files/example-file')
+            print(r)
 
+            ctadata.fetch_and_save_file(
+                r['path'], f'{tmpdir}/restored-file-example')
 
-def test_apiclient_list(start_service):
-    import ctadata
-    r = ctadata.list_dir("", downloadservice=start_service['url'])
 
-    print(r)
+@pytest.mark.timeout(30)
+def test_apiclient_upload_invalid_path(start_service, caplog):
+    with webdav_server():
+        import ctadata
+        ctadata.APIClient.downloadservice = start_service['url']
 
-    ctadata.APIClient.downloadservice = start_service['url']
+        with tempfile.TemporaryDirectory() as tmpdir:
+            subprocess.check_call(
+                ["dd", "if=/dev/random", f"of={tmpdir}/local-file-example",
+                 "bs=1M", "count=1"])
 
-    with pytest.raises(ctadata.api.StorageException):
-        ctadata.list_dir("blablabalfake")
+            with pytest.raises(ctadata.api.StorageException):
+                ctadata.upload_file(
+                    f'{tmpdir}/local-file-example', '../example-file')
 
-    r = ctadata.list_dir("lst")
 
-    n_dir = 0
-    n_file = 0
+@pytest.mark.timeout(30)
+def test_apiclient_upload_wrong(start_service, caplog):
+    with webdav_server():
+        import ctadata
+        ctadata.APIClient.downloadservice = start_service['url']
 
-    for entry in r:
-        print(entry)
-        if entry['type'] == 'directory':
-            n_dir += 1
-            print(ctadata.list_dir(entry['href']))
-        else:
-            n_file += 1
+        with tempfile.TemporaryDirectory() as tmpdir:
+            subprocess.check_call(
+                ["dd", "if=/dev/random", f"of={tmpdir}/local-file-example",
+                 "bs=1M", "count=1"])
 
-        if n_dir > 2 and n_file > 3:
-            break
+            with pytest.raises(ctadata.api.StorageException):
```

Comment:
is this exception going to be informative for the user? let's check how it looks to make sure it has the right comment